Рефакторинг

NullVerdict opened the pull request:
Коду уже 3 года, пришло время обновить его в соответствии с новыми возможностями Python. Также переделал Github Action, потому сборка занимает около 7 минут, в то время как текущая версия - около 1 минуты.

ValdikSS commented:
Утеряна совместимость с Python 3.7, перестало работать на Debian 10:

```
root@38617b4d617b:/work# python3 --version
Python 3.7.3
root@38617b4d617b:/work# cat /etc/debian_version 
10.13

File "torparse.pyz/tor_relay_scanner/scanner.py", line 72
return data.get("relays", []) if (data := self.grab(preferred_urls)) else []
                                        ^
```

Сборка для Windows использует Python 3.8.10 намеренно — это последняя версия, поддерживающая Windows 7.

>сборка занимает около 7 минут

Можно попробовать собирать в Docker-контейнере (чтобы не тратить время на удаление ненужных пакетов и установку дополнительных), а контейнер кешировать, как [здесь](https://github.com/bol-van/zapret/pull/407).

>Коду уже 3 года, пришло время обновить его в соответствии с новыми возможностями Python

Рефакторинг это хорошо, но только если это не ухудшает совместимость. Синтаксический сахар ради сахара на таком малом проекте ценой потери работы со старыми версиями — моветон.

SashaXser commented:
> Утеряна совместимость с Python 3.7, перестало работать на Debian 10:
> 
> ```
> root@38617b4d617b:/work# python3 --version
> Python 3.7.3
> root@38617b4d617b:/work# cat /etc/debian_version 
> 10.13
> 
> File "torparse.pyz/tor_relay_scanner/scanner.py", line 72
> return data.get("relays", []) if (data := self.grab(preferred_urls)) else []
>                                         ^
> ```
> 
> Сборка для Windows использует Python 3.8.10 намеренно — это последняя версия, поддерживающая Windows 7.
> 
> > сборка занимает около 7 минут
> 
> Можно попробовать собирать в Docker-контейнере (чтобы не тратить время на удаление ненужных пакетов и установку дополнительных), а контейнер кешировать, как [здесь](https://github.com/bol-van/zapret/pull/407).
> 
> > Коду уже 3 года, пришло время обновить его в соответствии с новыми возможностями Python
> 
> Рефакторинг это хорошо, но только если это не ухудшает совместимость. Синтаксический сахар ради сахара на таком малом проекте ценой потери работы со старыми версиями — моветон.

Хорошо, верну поддержку версии 3.7

ValdikSS commented:
Оформите нормально, по коммиту на изменение.

SashaXser commented:
> Оформите нормально, по коммиту на изменение.


Может, имеет смысл пересоздать с одним коммитом? ли как вы предлагаете передать изменения? Можно откатить и разделить на два отдельных коммита: первый — форматирование кода и совместимость с Python 3.7, второй — компиляция под 32-битную архитектуру.

ValdikSS commented:
Я предлагаю вам включать в коммит только то, что в нём написано. Одно изменение в каждом.
Сейчас у вас в `Совместимость с Python 3.7 и компиляция под 32-битную архитектуру` куча косметических изменений, вы pep8 прошлись?

SashaXser commented:
> Сейчас у вас в `Совместимость с Python 3.7 и компиляция под 32-битную архитектуру` куча косметических изменений, вы pep8 прошлись?

Это линтер Ruff. Я предлагаю откатить изменения и затем создать два отдельных коммита: в первом внесу изменения, а во втором выполню линтинг с помощью Ruff. Если нужно, могу временно отключить линтер.

ValdikSS commented:
Руководствуйтесь
https://www.kernel.org/doc/html/v4.10/process/submitting-patches.html#describe-your-changes
и
https://www.kernel.org/doc/html/v4.10/process/submitting-patches.html#separate-your-changes

SashaXser commented:
Теперь всё упорядочено и разбито по отдельным коммитам. Однако, на мой взгляд, эти правила больше подходят для управления крупными проектами, где много ревьюеров и пулл-реквестов, а сама кодовая база очень большая. В любом случае, это вопрос субъективный.

ValdikSS commented:
Не нужно revert'ить коммиты, это еще сильнее всё запутало. Сделайте с чистого листа, с head'а master'а. Для этого не нужно открывать еще один pull request, достаточно в этот же сделать `push --force`.

>Однако, на мой взгляд, эти правила больше подходят для управления крупными проектами, где много ревьюеров и пулл-реквестов

Я открываю ваш коммит и вместо того, чтобы просто взглянуть и оценить изменение (и принять его тут же, если оно маленькое и полезное) вынужден пробираться сквозь чащи кода, где одинарная кавычка заменена на двойную.
Я чрезвычайно загружен и пытаюсь уменьшить как могу своё время за компьютером.

ValdikSS commented:
Линтер, по моему мнению, большинство мест сделал хуже. Особенно в конце, где перечисления опций.

SashaXser commented:
Кажется, всё готово для мерджа. Если что-то упустил, дай знать, когда сможешь.

ValdikSS commented:
Сделайте, пожалуйста, ВСЕ логические изменения отдельными коммитами. Не нужно в `Рефакторинг` включать изменение github actions, чтобы потом его изменить еще одним. Если это рефакторинг, ломающий совместимость, а вы ее потом возвращаете, то это плохой pull request.

Вы пробовали билд .exe на Windows 7? Он работает? Версия должна быть именно 3.8.10, более новые из ветки 3.8 у меня не работали. Проверьте.

SashaXser commented:
> Сделайте, пожалуйста, ВСЕ логические изменения отдельными коммитами. Не нужно в `Рефакторинг` включать изменение github actions, чтобы потом его изменить еще одним. Если это рефакторинг, ломающий совместимость, а вы ее потом возвращаете, то это плохой pull request.
> 
> Вы пробовали билд .exe на Windows 7? Он работает? Версия должна быть именно 3.8.10, более новые из ветки 3.8 у меня не работали. Проверьте.

Не переживай, exe-билд точно работает, и он собран на версии 3.8.10. Завтра внесу изменения в два отдельных коммита: один для Python-файла, другой для Github Action. Если в будущем ты захочешь сам изменить Github Action, могу исключить его изменения из пулл-реквеста.

NullVerdict committed:
Рефакторинг scanner.py

Введена функция format_host для улучшения работы с форматированием адресов IPv6.

Использование f-строк для согласованного форматирования строк.

Заменена явная обработка исключений на contextlib для лучшего управления ресурсами при HTTP-запросах.

Упрощена фильтрация реле по порту, улучшена логика и ясность кода.

NullVerdict committed:
Рефакторинг GitHub Actions

Обновлены версии actions/checkout до v4 и actions/setup-python до v5.

Удалена ручная установка Wine и Python для Windows, теперь используются официальные действия для настройки Python.

Удалены лишние шаги по кешированию Python и скачиванию с сайта.

Сборка exe-файла для Windows теперь происходит на образе windows-latest, а не через Wine.

Оптимизированы шаги установки зависимостей и сборки артефактов.

ValdikSS commented:
* Часть комментариев в коде пропала
* Коммиты всё так же не разделены логически, я не смогу проверить такой большой объем переписанного кода

ValdikSS commented:
Нет ответа от автора.

NullVerdict commented:
> Нет ответа от автора.

Ответ автора: Этот PR никогда бы не приняли. Лучше создать что-то новое, чем переделывать этот скрипт на Python
